edgexfoundry/device-camera-go
# refactor(snap): edgex-snap-hooks related upgrade

This PR upgrades snap hooks to [2.3.0 ](https://github.com/canonical/edgex-snap-hooks/releases/tag/v2.3.0)with new function and refactor:
- Combine install hook and configure hook
- Use library function for copying directory
- Remove export bash command to process valid .env files
- Use Fatal for logging

Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-camera-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

1. Install and setup:
```
sudo snap set edgex-device-camera app-options=true
sudo snap set edgex-device-camera config.writable-loglevel=DEBUG
sudo snap set edgex-device-camera config.service-startupmsg="#test start up message#"
sudo snap start edgex-device-camera
```
2. Check the .env file is valid now without `export` command:
```
cat /var/snap/edgex-device-camera/current/config/device-camera/res/device-camera.env 

SERVICE_STARTUPMSG="#test start up message#"
WRITABLE_LOGLEVEL="DEBUG"
```
3. Validate env vars in `.env` have been exported successfully:
```
sudo snap logs -f edgex-device-camera

msg="#test start up message#"
level=DEBUG msg="xxx"
```
4. Validate all config files have been copied in right place:
```
tree /var/snap/edgex-device-camera/current/config
.
└── device-camera
    └── res
        ├── configuration.toml
        ├── device-camera.env
        ├── devices
        │   └── camera.toml
        └── profiles
            ├── camera-axis.yaml
            ├── camera-bosch.yaml
            └── camera.yaml

4 directories, 6 files


```

## New Dependency Instructions (If applicable)
